### PR TITLE
improve error message when include assertions fail

### PR DESCRIPTION
**Pull request by grosser**
`assert [1, 3].includes?(2)` fails with unhelpful "Asserting failed" message

`assert_includes [1, 3], 2` fails with "Expected [1, 3] to include 2" which makes it easier to debug and more obvious what went wrong

### CODE REVIEW

File: actionpack/test/controller/base_test.rb [hunk 1]
```diff
@@ -194,7 +194,7 @@ def test_url_helpers_does_not_become_actions
         get "account/overview"
       end
 
-      assert !@controller.class.action_methods.include?("account_overview_path")
+      refute_includes @controller.class.action_methods, "account_overview_path"
```
Comment (rafaelfranca) [inline actionpack/test/controller/base_test.rb]: `assert_not_includes`, we don't use and `refute` in our test suite.

File: actionpack/test/controller/helper_test.rb [hunk 2]
```diff
@@ -185,26 +185,26 @@ def test_all_helpers_with_alternate_helper_dir
     @controller_class.helper :all
 
     # helpers/abc_helper.rb should not be included
-    assert !master_helper_methods.include?(:bare_a)
+    refute_includes master_helper_methods, :bare_a
```
Comment (rafaelfranca) [inline actionpack/test/controller/helper_test.rb]: `assert_not_includes`

File: actionpack/test/controller/render_test.rb [hunk 3]
```diff
@@ -746,7 +746,7 @@ def test_head_with_symbolic_status
 
     get :head_with_symbolic_status, params: { status: "no_content" }
     assert_equal 204, @response.status
-    assert !@response.headers.include?("Content-Length")
+    refute_includes @response.headers, "Content-Length"
```
Comment (rafaelfranca) [inline actionpack/test/controller/render_test.rb]: `assert_not_includes`

File: actionpack/test/dispatch/cookies_test.rb [hunk 4]
```diff
@@ -68,7 +68,7 @@ def test_key_methods
   def test_write_doesnt_set_a_nil_header
     headers = {}
     request.cookie_jar.write(headers)
-    assert !headers.include?("Set-Cookie")
+    refute_includes headers, "Set-Cookie"
```
Comment (rafaelfranca) [inline actionpack/test/dispatch/cookies_test.rb]: `assert_not_includes`

File: actionpack/test/dispatch/cookies_test.rb [hunk 5]
```diff
@@ -1115,11 +1115,11 @@ def test_cookies_retained_across_requests
     assert_equal "david", cookies[:user_name]
 
     get :noop
-    assert !@response.headers.include?("Set-Cookie")
+    refute_includes @response.headers, "Set-Cookie"
```
Comment (rafaelfranca) [inline actionpack/test/dispatch/cookies_test.rb]: `assert_not_includes`

File: actionpack/test/dispatch/cookies_test.rb [hunk 6]
```diff
@@ -1115,11 +1115,11 @@ def test_cookies_retained_across_requests
...
     assert_equal "david", cookies[:user_name]
 
     get :noop
-    assert !@response.headers.include?("Set-Cookie")
+    refute_includes @response.headers, "Set-Cookie"
```
Comment (rafaelfranca) [inline actionpack/test/dispatch/cookies_test.rb]: `assert_not_includes`

File: actionview/test/template/form_helper_test.rb [hunk 7]
```diff
@@ -3120,7 +3120,7 @@ def test_fields_for_object_with_bracketed_name_and_index
   end
 
   def test_form_builder_does_not_have_form_for_method
-    assert !ActionView::Helpers::FormBuilder.instance_methods.include?(:form_for)
+    refute_includes ActionView::Helpers::FormBuilder.instance_methods, :form_for
```
Comment (rafaelfranca) [inline actionview/test/template/form_helper_test.rb]: `assert_not_includes`

File: actionview/test/template/test_case_test.rb [hunk 8]
```diff
@@ -209,7 +209,7 @@ class ATestHelperTest < ActionView::TestCase
     end
 
     test "no additional helpers should shared across test cases" do
-      assert !test_case.ancestors.include?(AnotherTestHelper)
+      refute_includes test_case.ancestors, AnotherTestHelper
```
Comment (rafaelfranca) [inline actionview/test/template/test_case_test.rb]: `assert_not_includes`

File: activemodel/test/cases/errors_test.rb [hunk 9]
```diff
@@ -125,7 +125,7 @@ def test_no_key
     person.errors[:foo]
     assert person.errors.empty?
     assert person.errors.blank?
-    assert !person.errors.include?(:foo)
+    refute_includes person.errors, :foo
```
Comment (rafaelfranca) [inline activemodel/test/cases/errors_test.rb]: `assert_not_includes`

File: actionview/test/template/test_case_test.rb [hunk 10]
```diff
@@ -161,7 +161,7 @@ class ViewAssignsTest < ActionView::TestCase
     test "view_assigns excludes internal ivars" do
       INTERNAL_IVARS.each do |ivar|
         assert defined?(ivar), "expected #{ivar} to be defined"
-        assert !view_assigns.keys.include?(ivar.to_s.tr("@", "").to_sym), "expected #{ivar} to be excluded from view_assigns"
+        refute_includes view_assigns.keys, ivar.to_s.tr("@", "").to_sym, "expected #{ivar} to be excluded from view_assigns"
```
Comment (rafaelfranca) [inline actionview/test/template/test_case_test.rb]: `assert_not_includes`

File: activemodel/test/cases/serializers/json_serialization_test.rb [hunk 11]
```diff
@@ -79,7 +79,7 @@ def setup
     assert_match %r{"name":"Konata Izumi"}, json
     assert_match %r{"age":16}, json
     assert_no_match %r{"awesome":true}, json
-    assert !json.include?(%("created_at":#{ActiveSupport::JSON.encode(Time.utc(2006, 8, 1))}))
+    refute_includes json, %("created_at":#{ActiveSupport::JSON.encode(Time.utc(2006, 8, 1))})
```
Comment (rafaelfranca) [inline activemodel/test/cases/serializers/json_serialization_test.rb]: `assert_not_includes`

File: activerecord/test/cases/adapters/postgresql/schema_authorization_test.rb [hunk 12]
```diff
@@ -100,10 +100,10 @@ def test_sequence_schema_caching
   end
 
   def test_tables_in_current_schemas
-    assert !@connection.tables.include?(TABLE_NAME)
+    refute_includes @connection.tables, TABLE_NAME
```
Comment (rafaelfranca) [inline activerecord/test/cases/adapters/postgresql/schema_authorization_test.rb]: `assert_not_includes`

**Review by rafaelfranca — CHANGES_REQUESTED** [review]
Looks good to me but could you change all `refute_includes` with `assert_not_includes`

**Comment by grosser** [comment]
why ?

using refute has the advantage of aligning nicely with assert

assert_includes a,b
refute_includes a,c

vs

assert_includes a,b
assert_not_includes a,c

**Comment by rafaelfranca** [comment]
We just don't like `refute` and don't want to use in our codebase.

**Comment by grosser** [comment]
hmm ok ... would you merge a PR to undef them ?

**Comment by grosser** [comment]
updated, looks good ?

**Comment by rafaelfranca** [comment]
I don't want to undef because people can still use in their applications.

**Review by rafaelfranca — APPROVED** [review]

**Comment by grosser** [comment]
can you also merge ? :)

**Comment by rafaelfranca** [comment]
Sure. I was waiting the build to finish 😄 

Thank you so much  :heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**